src/hooks/use-ordered-rows.ts

```diff
@@ -15,16 +18,19 @@ export function useOrderedRows<Row>(
       return rows;
     }
 
-    return [...rows].sort((a, b) => {
-      if (a[order.field] === b[order.field]) {
-        return 0;
+    return [...rows].sort(({ [order.field]: a }, { [order.field]: b }) => {
+      const x = order.direction === 'ascending' ? a : b;
+      const y = order.direction === 'ascending' ? b : a;
+
+      if (typeof x === 'string' && typeof y === 'string') {
+        return collator.compare(x, y);
       }
 
-      if (order.direction === 'ascending') {
-        return a[order.field] > b[order.field] ? 1 : -1;
+      if (x === y) {
+        return 0;
       }
 
-      return a[order.field] > b[order.field] ? -1 : 1;
+      return x > y ? 1 : -1;
```

Comment:
As someone who has been working a lot with floats recently, I feel tempted to point out `NaN` issues (seriously though, don't worry about that!)

Reply:
😅